smarts/sstudio/types.py

```diff
@@ -813,6 +813,36 @@ def to_geometry(self, road_map: Optional[RoadMap] = None) -> Polygon:
             poly = shapely_rotate(poly, self.rotation, use_radians=True)
         return shapely_translate(poly, xoff=x, yoff=y)
 
+@dataclass(frozen=True)
+class ConfigurableZone(Zone):
+    """A descripter that defines a specific configurableZone defined by user"""
+
+    ext_coordinates: List[Tuple[float, float]]
+    """external coordinates of the polygon"""
```

Comment:
Please describe how many points are valid for what kind of shape is desired. 

- <2 points causes an error
- 2 points generates a box
- \>2 points generates a shape